modules/core/src/main/java/org/apache/ignite/internal/processors/query/TableInformation.java

```diff
@@ -0,0 +1,226 @@
+        this.keyAlias = keyAlias;
+        this.valAlias = valAlias;
+        this.keyTypeName = keyTypeName;
+        this.valTypeName = valTypeName;
+    }
+
+    /**
+     * @param schemaName Schema name.
+     * @param tblName Table name.
+     * @param tblType Table type.
+     */
+    public TableInformation(String schemaName, String tblName, String tblType){
+        this.schemaName = schemaName;
+        this.tblName = tblName;
+        this.tblType = tblType;
+
+        this.cacheGrpId = -1;
+        this.cacheGrpName = null;
+        this.cacheId = -1;
+        this.cacheName = null;
+        this.affinityKeyCol = null;
+        this.keyAlias = null;
+        this.valAlias = null;
+        this.keyTypeName = null;
+        this.valTypeName = null;
+
+    }
+
+    /**
+     * @return Schema name.
+     */
+    public String schemaName() {
+        return schemaName;
+    }
+
+    /**
+     * @return Table name.
+     */
+    public String tableName() {
+        return tblName;
+    }
+
+    /**
+     * @return Table type.
+     */
+    public String tableType() {
+        return tblType;
+    }
+
+    /**
+     * @return Cache Group id or {@code -1} if not applicable.
+     */
+    public int cacheGrpId() {
+        return cacheGrpId;
+    }
+
+    /**
+     * @return Cache group name or {@code null} if not applicable.
+     */
+    public String cacheGrpName() {
+        return cacheGrpName;
+    }
+
+    /**
+     * @return Cache id or {@code -1} if not applicable.
+     */
+    public int cacheId() {
+        return cacheId;
+    }
+
+    /**
+     * @return Cache name or {@code null} if not applicable.
+     */
+    public String cacheName() {
+        return cacheName;
+    }
+
+    /**
+     * @return Affinity key column name or {@code null} if not applicable.
+     */
+    public String affinityKeyColumn() {
+        return affinityKeyCol;
+    }
+
+    /**
+     * @return Key alias or {@code null} if not applicable.
+     */
+    public String keyAlias() {
+        return keyAlias;
+    }
+
+    /**
+     * @return Value alias or {@code null} if not applicable.
+     */
+    public String valueAlias() {
+        return valAlias;
+    }
+
+    /**
+     * @return Key type name or {@code null} if not applicable.
+     */
+    public String keyTypeName() {
+        return keyTypeName;
+    }
+
+    /**
+     * @return Value type name or {@code null} if not applicable.
+     */
+    public String valueTypeName() {
+        return valTypeName;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean equals(Object o) {
+        if (this == o)
+            return true;
+
+        if (o == null || getClass() != o.getClass())
+            return false;
+
+        TableInformation tblInfo = (TableInformation)o;
+
+        return F.eq(schemaName, tblInfo.schemaName) && F.eq(tblName, tblInfo.tblName);
+    }
+
+    /** {@inheritDoc} */
+    @Override public int hashCode() {
+        return Objects.hash(schemaName, tblName);
```

Comment:
Don't we consider `Objects.hash` as a bad practice?

Reply:
yep